### DISCUSSION TITLE
[WASM] AOT no longer builds with latest UNO bits

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#10485] [Q&A] [answered]
I used to build AOT like
`MSBuild.exe Unusual.Wasm.sln /restore /nologo /p:Configuration=Release /t:publish /v:minimal`

This no longer works as I get errors as per 1st screenshot.

Not knowing what I'm going here I tried to run the suggested command in an elevated PowerShell in different variations. None of them worked (2nd screenshot).

What am I doing wrong? How could I build WASM AOT as before? Is there any documentation available?

<img width="867" alt="image" src="https://user-images.githubusercontent.com/26179242/202704414-8c246c5f-d6df-4824-b7bc-53291d0e81fc.png">

<img width="1154" alt="image" src="https://user-images.githubusercontent.com/26179242/202705283-f3d5cd2c-03a7-4950-a539-26bd2309e0fd.png">

**Comment by jeromelaban** [comment]
You'll need to only run what's inside the quotes.

**Reply by DierkDroth** [reply]
@jeromelaban unfortunately that didn't work. 

I'm still getting the same error after running the command as per sceenshot below

<img width="675" alt="image" src="https://user-images.githubusercontent.com/26179242/202836083-c48ace8a-5b22-461f-834c-3549bc7c6840.png">

**Reply by jeromelaban** [reply]
Once you have run this, you need to restart visual studio.

**Reply by DierkDroth** [reply]
@jeromelaban thanks for your feedback.

To clarify: I'm not running the MSBuild command above from VS but it in a batch (.BAT) file. 

Nevertheless I shutdown VS, executed the PowerShell job again and still am getting the same error.

What am I missing?

**Reply by jeromelaban** [reply]
I don't understand what you mean by running in a batch file, this should run directly from an elevated command propmt.

To validate that the command worked, try running `Get-ExecutionPolicy` in poweshell prompt. If it does not show `RemoteSigned` the command above did not work, or was not set in the right environment for the current user.

**Reply by DierkDroth** [reply]
@jeromelaban please see below.
I also attached the batch (.BAT) file I was refering to. BTW: for now I'm running it by double click right from Windows explorer. CI/CD comes later...

Please let me know if you needed anything else. Thanks

<img width="675" alt="image" src="https://user-images.githubusercontent.com/26179242/203075349-4325b2f4-fd57-4d49-ab86-33fe881dfce6.png">

[Make.zip](https://github.com/unoplatform/uno/files/10057183/Make.zip)

**Reply by jeromelaban** [reply]
Thank you. Can you provide a binlog of your failing build?

**Reply by DierkDroth** [reply]
@jeromelaban pls see attached

[msbuild.zip](https://github.com/unoplatform/uno/files/10058164/msbuild.zip)

**Reply by jeromelaban** [reply]
Thanks, make sure [to create a binlog](https://platform.uno/docs/articles/uno-builds-troubleshooting.html).

**Reply by DierkDroth** [reply]
@jeromelaban here you go...

[msbuild.zip](https://github.com/unoplatform/uno/files/10058644/msbuild.zip)

**Reply by jeromelaban** [reply]
Thanks. For some reason your visual studio is not using the right powershell environment, could be related to your windows edition.

Can you try `Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope LocalMachine` ?

**Reply by DierkDroth** [reply]
@jeromelaban unfortunately it's still not working. 
Please see attached binlog

Also: executing `Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope LocalMachine` came with some serious security warnings (I executed it regardless for that test).

How could I revert to 'reasonable defaults'? Thanks

[msbuild.zip](https://github.com/unoplatform/uno/files/10064351/msbuild.zip)

**Reply by jeromelaban** [reply]
Thanks for the update. I do not have anything else for you to try, it seems that your environment forces scripts to be fully signed for some unknown reason. You may want to try some of [the options here](https://winbuzzer.com/2020/07/10/how-to-enable-powershell-scripts-in-windows-10-via-powershell-execution-policy-xcxwbt/).
 
You can restore using `Set-ExecutionPolicy -ExecutionPolicy AllSigned -Scope LocalMachine`

**Reply by DierkDroth** [reply]
@jeromelaban well, this is quite an unpleasant surprise and basically would be a showstopper to all my UNO efforts.

Let me try to understand where we're at:
- I was able to build UNO WASM AOT with .NET 6. What exactly has changed with .NET 7 and what makes it breaking now?
- what exact step fails and why - according your understanding?
- would this be an issue which Blazor AOT would run into as well or are we looking at a UNO specific issue here?
- why do we have to deal with PowerShell in the first place? Is there any alternative?
- I would think that what you labeled as 'my environment' should be pretty standard to lots of your potential or actual customers. AFAIK there is nothing special with 'my environment'. In fact, I'm trying to keep it as simple as possible
- pls understand that the innerts of all this is really nothing I have any expertise in. Pls be patient to provide me a chance to analyze the problem before I had to give up

**Reply by jeromelaban** [reply]
All users that we know have been using this version of Uno bootstrapper without issues; you're the first one to encounter this. Most of our users are going for Windows Pro or enterprise, which don't show this issue (if the issue really is related to it).

Powershell interactions are not optional, and we do not have a way to validate changes to your environment. This something you can do by testing what is provided by the link above. Try out the steps and let us know if it helps, particularly the one about "developer mode".

**Comment by DierkDroth** [comment, marked as answer]
@jeromelaban I was able to resolve the issue. AOT builds now. Thanks for your support

On Windows 11:
Control panel -> System -> Privacy & Security -> For developers -> PowerShell -> Change execution policy to allow... -> ON (screenshot below)

<img width="753" alt="image" src="https://user-images.githubusercontent.com/26179242/203399778-d29b5414-7f5b-4c3f-a00b-a19db6f4c064.png">
